x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/utils/get_mode_display_text.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ADMIN, READ_ONLY, READ_WRITE, SEARCH, SEARCH_DISPLAY, WRITE_ONLY } from '../constants';
+import { IApiToken } from '../types';
+
+export const getModeDisplayText = (apiToken: IApiToken): string => {
+  const { read = false, write = false, type } = apiToken;
+
+  switch (type) {
+    case ADMIN:
+      return '--';
+    case SEARCH:
```

Comment:
Not a change request, just me musing out loud - I wonder if we should make ADMIN, SEARCH, and PRIVATE an enum.

I've recently really grown to like enums after reviewing Byron's PRs, and it might make reading a little easier? (this is pretty subjective though, I definitely don't feel super strongly about it and would love your opinion!)

```ts
export enum ApiTokenTypes {
  Admin: 'admin',
  Private: 'private',
  Search: 'search',
}

switch (type) {
  case ApiTokenTypes.Admin:
    return '--';
  case ApiTokenTypes.Search:
    return SEARCH_DISPLAY;
  // ...etc
}
```
Nah? Is it overkill?

Reply:
No that's not overkill. I was actually thinking the same thing but I didn't bother with it at the time. But I'm going in to make changes anyway, so I'll make the switch.

Reply:
4917cbb

Reply:
30c52a71fecdbd5f5270709820c8088af8804dfa

Reply:
I totally underestimated how many files that change would affect 😅 I think it's a net readability gain though - maybe a few lines/couple places are slightly worse or more verbose (mostly due to # of characters) but I think it's a worthwhile tradeoff. The type specificity in particular I think is nice - it might be extra dev friction to have to be super specific, but I think it's growing on me.